### PR TITLE
ostreehub: allow delta files and 'summary' upload

### PR DESCRIPTION
**Pull request by mike-sul**
Signed-off-by: Mike Sul <mike.sul@foundries.io>

### CODE REVIEW

File: pkg/fiopush/push.go [hunk 1]
```diff
@@ -79,6 +79,8 @@ var (
 		"./objects/",
 		"./config",
 		"./refs/",
+		"./deltas/",
+		"./summary",
```
Comment (doanac) [inline pkg/fiopush/push.go:83]: I don't think we are going to want summary control from the client side in this way. I'd say we drop this until we know we need it

File: main.go [hunk 2]
```diff
@@ -26,6 +26,13 @@ func main() {
 	log.Println("bucket =", *bucket)
 	log.Println("upload worker number =", *workerNumb)
 
+	if *workerNumb == 42 {
```
Comment (doanac) [inline main.go:29]: you need to drop this hack out. it was part of the WIP

File: pkg/oshub/upload.go [hunk 3]
```diff
@@ -224,3 +241,30 @@ func upload(objectName string, object *RepoFile, srcFilePath string) *uploadStat
...
+	// GCS bucket layout
+	// <bucket-name>/<factory-sha256>/
+	//									[%2x]/ for objects,
+	//									"deltas/" for delta objects
+	//									"summary" file at the root
+	if strings.HasPrefix(object, "./objects/") {
+		return object[len("./objects/")-1:], nil
+	}
+	if strings.HasPrefix(object, "./deltas/") {
+		return object, nil
```
Comment (doanac) [inline pkg/oshub/upload.go:264]: I'm not sure where in the code it should fit, but we should call `RegenerateSummary(factoryPrefix)` after a `deltas/..../superblock` gets uploaded

File: pkg/fiopush/push.go (hunk 1 above)
Comment (mike-sul) [reply pkg/fiopush/push.go:83]: Yep, that's what I thought.

File: pkg/oshub/upload.go (hunk 3 above)
Comment (mike-sul) [reply pkg/oshub/upload.go:264]: I'll think about it, and then let's discuss it later today. There are a few options.

Comment (doanac) [reply pkg/oshub/upload.go:264]: the more i think about this. i bet i know what you are thinking: we could have a "put" API to regenerate static deltas. that would be pretty handy.

Comment (mike-sul) [reply pkg/oshub/upload.go:264]: You are reading my mind :)

**Comment by mike-sul** [comment]
@doanac I addressed the comments. The new handler to regenerate 'summary' will go into a new PR.

File: pkg/oshub/upload.go [hunk 4]
```diff
@@ -51,6 +51,13 @@ var (
 		bucketName string
 		workerNumb int
 	}
+
+	repoObjectsFilterIn = []string{
+		// This is what is synced with GCS bucket
+		"./objects/",
+		"./deltas/",
+		"./summary",
```
Comment (doanac) [inline pkg/oshub/upload.go:59]: think you missed this reference when refactoring.

Comment (mike-sul) [reply pkg/oshub/upload.go:59]: I left it for a purpose. I removed 'summary' from fiopush and the PUT/GET handler so a user won't be able to update it, but I left it here so we still can update it internally we might need it for the future summary regeneration.

**Review by doanac — APPROVED** [review]

**Pull request merged**